### ISSUE TITLE
MaxPool op does not work as expected when ceil_mode=True and padding=1.

### ISSUE THREAD

**Issue by fatcat-z** [microsoft/onnxruntime#15446]
### Describe the issue

In some case, like the scripts provided below to repro this issue, MaxPool op could not work as expected.

### To reproduce

Python code to repro this issue:

```python
# import torch
import onnxruntime

## Only when ceil_mode=True and padding>0, the bug occurs.
model = torch.nn.MaxPool3d(kernel_size=3, stride=None, padding=1, dilation=1, return_indices=False, ceil_mode=True)
x = torch.randn(1, 2, 3, 6, 5)

out_torch = model(x)

# torch.onnx.export(model, (x, ), 'max_pool3d_model.onnx')

# sess = onnxruntime.InferenceSession('max_pool3d_model.onnx')
model_text = """
<
   ir_version: 7,
   opset_import: ["" : 14],
   producer_name: "pytorch",
   producer_version: "2.1.0"
>
torch_jit (float[1,2,3,6,5] MaxPool_0) => (float[1,2,2,3,3] output) {
   output = MaxPool <ceil_mode = 1, kernel_shape = [3, 3, 3], pads = [1, 1, 1, 1, 1, 1], strides = [3, 3, 3]> (MaxPool_0)
}
"""
model = onnx.parser.parse_model(model_text)

sess = onnxruntime.InferenceSession(model.SerializeToString())
out_ort = sess.run(None, {sess.get_inputs()[0].name: x.numpy()})

out_ort = torch.tensor(out_ort[0])
out_torch = (
    out_torch
    if isinstance(out_torch, torch.Tensor)
    else torch.tensor(out_torch)
)

torch.testing.assert_close(out_ort, out_torch)

print("===== End =====")
```

#### Error

```
AssertionError: The values for attribute 'shape' do not match: torch.Size([1, 2, 2, 3, 3]) != torch.Size([1, 2, 2, 3, 2]).
```

### Platform

Linux

### OS Version

Ubuntu 20.04.4 LTS

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.14.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by wschin** [comment]
I only found MaxPoolV8 in CPU EP, so it could be a missing op.
```cpp
class MaxPoolV8 : public OpKernel, public PoolBase {

 template <typename T>
  struct ComputeHelper {
    Status operator()(const MaxPoolV8* inst, OpKernelContext* context) const {
      return inst->ComputeImpl<T>(context);
    }
  };

 public:
  MaxPoolV8(const OpKernelInfo& info) : OpKernel(info), PoolBase(info) {}
  Status Compute(OpKernelContext* context) const override;
 private:
  template <typename T>
  Status ComputeImpl(OpKernelContext* context) const;
};
```

**Comment by BowenBao** [comment]
Comment says below, it does support MaxPool up to latest version 12.
```c++
// For maxpool v8 and beyond
// version 8: Added storage_order And Indices
// version 10: Added ceil_mode
// version 11: Added dilations
// version 12: Added int8/uint8 support
class MaxPoolV8 : public OpKernel, public PoolBase {
```

Looking at the output shapes again, it shows rather that **torch** output shape is incorrect.

**Comment by BowenBao** [comment]
torch is performing special undocumented behavior under ceil_mode.

https://github.com/pytorch/pytorch/blob/624257890e89b3851e0cc0ae1af96b47c09b8055/aten/src/ATen/native/Pool.h#L49-L55

Not much we can do.

**Comment by BowenBao** [comment]
This is not only restricted to MaxPool3d. It repros under `ceil_mode=True` and certain combination of `kernel_size`, `pad` and `input_shape`. Below is a repro for MaxPool2d.

```python
import torch
import onnxruntime
import onnx

## Only when ceil_mode=True and padding>0, the bug occurs.
model = torch.nn.MaxPool2d(kernel_size=3, stride=None, padding=1, dilation=1, return_indices=False, ceil_mode=True)
x = torch.randn(1, 2, 3, 5)

out_torch = model(x)
# torch.onnx.export(model, (x, ), 'max_pool3d_model.onnx')

# sess = onnxruntime.InferenceSession('max_pool3d_model.onnx')
model_text = """
<
   ir_version: 7,
   opset_import: ["" : 14],
   producer_name: "pytorch",
   producer_version: "2.1.0"
>
torch_jit (float[1,2,3,5] MaxPool_0) => (float[1,2,2,3] output) {
   output = MaxPool <ceil_mode = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [3, 3]> (MaxPool_0)
}
"""
model = onnx.parser.parse_model(model_text)

sess = onnxruntime.InferenceSession(model.SerializeToString(), providers=['CPUExecutionProvider'])
out_ort = sess.run(None, {sess.get_inputs()[0].name: x.numpy()})

out_ort = torch.tensor(out_ort[0])
out_torch = (
    out_torch
    if isinstance(out_torch, torch.Tensor)
    else torch.tensor(out_torch)
)

torch.testing.assert_close(out_ort, out_torch)

print("===== End =====")
```

**Comment by justinchuby** [comment]
@titaiwangms

**Comment by shubhambhokare1** [comment]
Seems like this issue has been fixed based on [#5711](https://github.com/onnx/onnx/issues/5711)

Closing for now
